django/db/models/fields/tuple_lookups.py

```diff
@@ -338,6 +342,9 @@ def process_rhs(self, compiler, connection):
                 )
             )
 
+        if not result:
+            raise EmptyResultSet
+
```

Comment:
Notice how this `process_rhs` was missing the matching logic from its superclass

https://github.com/django/django/blob/c7ff347c641f2f97fa9f2f7d182982f789a211b4/django/db/models/lookups.py#L527-L536